File: Leetcode/Grind 75 INTERVIEW PRACTICE/twoSum.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        hashset={} # set to keep nums's element and position
        for i,n in enumerate(nums):
            diff=target-n # the remain value
            if diff in hashset: # if the remain value was lay in previous position
                return [hashset[diff],i] # return the previous position and current position
                
            hashset[n]=i #if not, keep the position record for current number

File: Leetcode/Grind 75 INTERVIEW PRACTICE/twoSum.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        unordered_map<int,int> prevElement;

        for (int i=0;i<nums.size();i++)
        {
            int dif=target-nums[i];
            if(prevElement.count(dif))
            {
                return {prevElement[dif],i};
            }

            prevElement[nums[i]]=i;

        }

        return {0,0};
    }
};
